processor/metricstransformprocessor/metrics_transform_processor_test.go

```diff
@@ -0,0 +1,306 @@
+			outLabels:  outLabels,
+		},
+		{
+			name:       "metric_label_invalid_update",
+			metricName: "metric1",
+			action:     "update",
+			operations: []Operation{invalidUpdateLabelOperation},
+			inMN:       inMetricNames,
+			outMN:      inMetricNames,
+			inLabels:   inLabels,
+			outLabels:  inLabels,
+		},
+		// INSERT
+		{
+			name:       "metric_name_insert",
+			metricName: "metric1",
+			action:     "insert",
+			newName:    "metric1/new",
+			inMN:       inMetricNames,
+			outMN:      outMetricNamesInsert,
+		},
+		{
+			name:       "metric_label_update_with_metric_insert",
+			metricName: "metric1",
+			action:     "insert",
+			newName:    "metric1/new",
+			operations: []Operation{validUpateLabelOperation},
+			inMN:       inMetricNames,
+			outMN:      outMetricNamesInsert,
+			inLabels:   inLabels,
+			outLabels:  outLabels,
+		},
+	}
+)
+
+func TestMetricsTransformProcessor(t *testing.T) {
+	for _, test := range standardTests {
+		t.Run(test.name, func(t *testing.T) {
+			// next stores the results of the aggregation metric processor
+			next := &etest.SinkMetricsExporter{}
+			cfg := &Config{
+				ProcessorSettings: configmodels.ProcessorSettings{
+					TypeVal: typeStr,
+					NameVal: typeStr,
+				},
+				MetricName: test.metricName,
+				Action:     test.action,
+				NewName:    test.newName,
+				Operations: test.operations,
+			}
+			amp, err := newMetricsTransformProcessor(next, cfg)
+			assert.NotNil(t, amp)
+			assert.Nil(t, err)
+
+			caps := amp.GetCapabilities()
+			assert.Equal(t, false, caps.MutatesConsumedData)
+			ctx := context.Background()
+			assert.NoError(t, amp.Start(ctx, nil))
+
+			md := consumerdata.MetricsData{
+				Metrics: make([]*metricspb.Metric, len(test.inMN)),
+			}
+
+			// construct metrics data to feed into the processor
+			for idx, in := range test.inMN {
+				labels := make([]*metricspb.LabelKey, 0)
+				if in == test.metricName {
+					for _, l := range test.inLabels {
+						labels = append(
+							labels,
+							&metricspb.LabelKey{
+								Key: l,
+							},
+						)
+					}
+				}
+				md.Metrics[idx] = &metricspb.Metric{
+					MetricDescriptor: &metricspb.MetricDescriptor{
+						Name:      in,
+						LabelKeys: labels,
+					},
+				}
+			}
+
+			// process
+			cErr := amp.ConsumeMetrics(
+				context.Background(),
+				pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
+					{},
+					md,
+					{
+						Metrics: []*metricspb.Metric{},
+					},
+				}),
+			)
+			assert.Nil(t, cErr)
+
+			// get and check results
+			got := next.AllMetrics()
+			require.Equal(t, 1, len(got))
+			gotMD := pdatautil.MetricsToMetricsData(got[0])
+			require.Equal(t, 3, len(gotMD))
+			assert.EqualValues(t, consumerdata.MetricsData{}, gotMD[0])
+			require.Equal(t, len(test.outMN), len(gotMD[1].Metrics))
+			for idx, out := range gotMD[1].Metrics {
+				// check name
+				assert.Equal(t, test.outMN[idx], out.MetricDescriptor.Name)
+
+				//check labels
+				if test.action == Insert {
```

Comment:
This test could grow quite large. May be okay as is, but I'd consider splitting it up by different types of actions.

Reply:
Just updated this to a simpler format, which took out around 10 lines of the repetitive code here. If I see it grows larger, I can further modulate it by setting up helper functions, which will handle these based on actions